graylog2-server/src/main/java/org/graylog2/database/dbcatalog/DbEntitiesScanner.java

```diff
@@ -33,21 +35,29 @@ public class DbEntitiesScanner implements Provider<DbEntitiesCatalog> {
 
     private final String[] packagesToScan;
 
+    private final ClassLoader chainingClassLoader;
+
     @SuppressWarnings("unused")
-    public DbEntitiesScanner() {
+    @Inject
+    public DbEntitiesScanner(final ChainingClassLoader chainingClassLoader) {
+        this.chainingClassLoader = chainingClassLoader;
         this.packagesToScan = new String[]{"org.graylog2", "org.graylog"};
     }
 
-    @SuppressWarnings("unused")
     DbEntitiesScanner(String[] packagesToScan) {
+        this.chainingClassLoader = null;
         this.packagesToScan = packagesToScan;
+
     }
 
     @Override
     public DbEntitiesCatalog get() {
         final ConfigurationBuilder configuration = new ConfigurationBuilder()
                 .forPackages(packagesToScan)
                 .setScanners(Scanners.TypesAnnotated);
+        if (chainingClassLoader != null) {
+            configuration.setClassLoaders(new ClassLoader[]{chainingClassLoader});
+        }
```

Comment:
Apparently, the classloader needs to be provided with each package to scan. Otherwise, it won't pick up the correct jars. The following seems to work for me:

```suggestion
        final ConfigurationBuilder configuration = new ConfigurationBuilder();

        if (chainingClassLoader != null) {
            configuration.setClassLoaders(new ClassLoader[]{chainingClassLoader});
            Stream.of(packagesToScan).forEach(pkg -> {
                configuration.forPackage(pkg, chainingClassLoader);
            });
        } else {
            configuration.forPackages(packagesToScan);
        }

        configuration.setScanners(Scanners.TypesAnnotated);
```

Reply:
It works! Thank you!